Python/Product/TestAdapter.Executor/Pytest/PytestTestDiscoverer.cs

```diff
@@ -28,17 +28,24 @@
 using Newtonsoft.Json;
 
 namespace Microsoft.PythonTools.TestAdapter.Pytest {
-    internal class TestDiscovererPytest : IPythonTestDiscoverer {
-        private readonly PythonProjectSettings _settings;
+
+    /// <summary>
+    /// Note even though we specify  [DefaultExecutorUri(PythonConstants.PytestExecutorUriString)] we still get all .py source files
+    /// from all testcontainers.  
+    /// </summary>
+    [FileExtension(".py")]
+    [DefaultExecutorUri(PythonConstants.PytestExecutorUriString)]
+    public class PytestTestDiscoverer : PythonTestDiscoverer {
         private IMessageLogger _logger;
         private static readonly string DiscoveryAdapterPath = PythonToolsInstallPath.GetFile("PythonFiles\\testing_tools\\run_adapter.py");
 
-        public TestDiscovererPytest(PythonProjectSettings settings) {
```

Comment:
moving settings from a constructor param to a function param. Also simplifies testing by not having to build a settings object